src/cpx_genericcallbacks.jl

```diff
@@ -0,0 +1,260 @@
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetcandidateray(callback_data::CallbackContext, x::Vector{Cdouble}, begin_index::Int, end_index::Int)
+    return_status = @cpx_ccall(callbackgetcandidateray,
+                Cint,
+                (Ptr{Cvoid}, Ptr{Cdouble}, Cint, Cint),
+                callback_data.context, x, begin_index - 1, end_index - 1)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetfunc(model::Model, context_id::Clong, callback_func::Function)
+    c_callback = @cfunction(
+        callback_wrapper, Cint, (Ptr{Cvoid}, Clong, Ptr{Cvoid})
+    )
+    user_handle = (model, callback_func)::Tuple{Model, Function}
+    ret = @cpx_ccall(callbackgetfunc,
+        Cint,
+        (Ptr{Cvoid}, Ptr{Cvoid}, Ref{Clong}, Ptr{Cvoid}, Any),
+        model.env, model.lp, context_mask, c_callback, user_handle)
+    if ret != 0
+        throw(CplexError(model.env, ret))
+    end
+    model.callback = user_handle
+    return ret
+end
+
+function cbgetincumbent(callback_data::CallbackContext, x::Vector{Cdouble}, begin_index::Int, end_index::Int, obj::Cdouble)
+    return_status = @cpx_ccall(callbackgetincumbent,
+                    Cint,
+                    (Ptr{Cvoid}, Ptr{Cdouble}, Cint, Cint, Ref{Cdouble}),
+                    callback_data.context, x, begin_index - 1, end_index - 1, obj)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetinfodbl(callback_data::CallbackContext, cbinfo::CbInfo, data_p::Cdouble)
+    return_status = @cpx_ccall(callbackgetinfodbl,
+                    Cint,
+                    (Ptr{Cvoid}, Cint, Ref{Cdouble}),
+                    callback_data.context, cbinfo, data_p)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetinfoint(callback_data::CallbackContext, cbinfo::CbInfo, data_p::Int)
+    return_status = @cpx_ccall(callbackgetinfoint,
+                    Cint,
+                    (Ptr{Cvoid}, Cint, Ref{Cint}),
+                    callback_data.context, cbinfo, data_p)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetinfolong(callback_data::CallbackContext, cbinfo::CbInfo, data_p::Clong)
+    return_status = @cpx_ccall(callbackgetinfolong,
+                    Cint,
+                    (Ptr{Cvoid}, Cint, Ref{Clong}),
+                    callback_data.context, cbinfo, data_p)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbrejectcandidate(callback_data::CallbackContext, rcnt::Int, nzcnt::Int, rhs::Cdouble, sense::Char, rmatbeg::Vector{Int}, rmatind::Vector{Int}, rmatval::Vector{Cdouble})
```

Comment:
Since those parameters are related to sparse representation of constraints, I changed `rmatbeg` to `sparse_index_begin`, `rmatind` to `sparse_index`, and `rmatval` to `sparse_index_value`. Also made changes to some other ambiguous names.